logic/atrium-logic-jvm/src/main/kotlin/ch/tutteli/atrium/logic/impl/DefaultPathAssertions.kt

```diff
@@ -194,4 +193,31 @@ class DefaultPathAssertions : PathAssertions {
                     container.resolve(entry).collect { _logicAppend { exists(NOFOLLOW_LINKS) } }
                 }
         ).build()
+
+    override fun <T : Path> isEmptyDirectory(container: AssertionContainer<T>): Assertion =
+        if (container.isDirectory().holds()) {
+            container.changeSubject.unreported {
+                it.runCatchingIo { Files.list(it).findAny() }
+            }
+                .let { expectResult ->
+                    assertionBuilder.descriptive.withTest(expectResult)
+                    { it is Success && it.value.isEmpty }
+                        .withFailureHintBasedOnDefinedSubject(expectResult) {
+                            when (it) {
+                                is Success ->
+                                    assertionBuilder.descriptive.failing
+                                        .withDescriptionAndRepresentation(
+                                            DIRECTORY_HAS_FILE,
+                                            it.value.orElseThrow().toString()
```

Comment:
leave out the toString, it is the responsibility of the ObjectFormatter how Paths are formatted:
```suggestion
                                            it.value.orElseThrow()
```